File: src/employeemanager/model/Department.java
package employeemanager.model;

import java.util.ArrayList;
import java.util.List;

public class Department {
    private int departmentId;
    private String name;
    private final List<Employee> employees = new ArrayList<>();

    public Department(int departmentId, String name) {
        this.departmentId = departmentId;
        this.name = name;
    }

    public int getDepartmentId() {
        return departmentId;
    }

    public String getName() {
        return name;
    }

    public void addEmployee(Employee employee) {
        if (!employees.contains(employee)) {
            employees.add(employee);
            employee.setDepartment(this);
        }
    }

    public List<Employee> getEmployees() {
        return new ArrayList<>(employees);
    }
}

File: Part 3/EmployeeManager/models/Department.cs
using System.Collections.Generic;

namespace EmployeeManager.Models
{
    public class Department
    {
        public int DepartmentId { get; set; }
        public string Name { get; set; }
        public readonly List<Employee> _employees = new();

        public Department(int departmentId, string name)
        {
            DepartmentId = departmentId;
            Name = name;
        } 

        public void AddEmployee(Employee employee)
        {
            if (!_employees.Contains(employee))
            {
                _employees.Add(employee);
                employee.Department = this;
            }
        }

        public IReadOnlyList<Employee> Employees => _employees.AsReadOnly();
    }
}